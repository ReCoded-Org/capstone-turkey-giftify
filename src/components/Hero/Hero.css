/*Button is not working from react-bootstrap*/

HTML {
  background-color: #0077b6;
  font-family: Roboto;
}

@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");

.App {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
  background-color: #0077b6;
}

.hero {
  background-color: #ade8f4;
  width: 75%;
  border-radius: 20px;
  margin-left: 15%;
}

.upperPart {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  align-content: stretch;
  margin-top: 5%;
}

.texts {
  width: 65%;
  padding-left: 5%;
}

.texts > h2 {
  font-size: 60px;
  color: #3f3b3b;
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  line-height: 70px;
  letter-spacing: -0.7px;
}

.texts > p {
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 20px;
  line-height: 30px;
  color: #3f3b3b;
  width: 70%;
}

.heroImg {
  margin-right: 5%;
}

.heroImg > img {
  border-radius: 40px;
}

.giftifyButton {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
}

.giftifyButton > a {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
  text-decoration: none;
  padding: 16px 72px;
}

.giftifyButton > a > button {
  background-color: #023e8a;
  border-radius: 0.75rem;
  color: #ffffff;
  font-family: Roboto;
  padding: 10% 50%;
  font-size: 20px;
  font-style: normal;
  font-weight: normal;
  clear: both;
  display: inline-block;
  white-space: nowrap;
}

/*tablet*/
@media (min-width: 768px) and (max-width: 1024px) {
  .hero {
    width: 100%;
    border-radius: 0%;
  }

  .texts > p {
    width: 80%;
  }
}

/*mobile*/
@media only screen and (min-width: 120px) and (max-width: 767px) {
  .hero {
    width: 100%;
    border-radius: 0%;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: center;
    align-items: center;
    align-content: stretch;
  }

  .heroImg {
    display: none;
  }

  .texts > h2 {
    font-family: Roboto;
    font-size: 40px;
    font-style: normal;
    font-weight: 500;
    line-height: 48px;
    letter-spacing: -0.4000000059604645px;
    text-align: left;
  }

  .texts {
    width: 80%;
    padding-left: 20%;
  }
}
