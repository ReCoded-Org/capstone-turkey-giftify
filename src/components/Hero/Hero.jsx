import React from "react";
import "./Hero.css";
import { BrowserRouter as Router, Link } from "react-router-dom";
import { Button } from "react-bootstrap";
//Button is not working from react-bootstrap

function Hero() {
  return (
    <div className="hero">
      <div className="upperPart">
        <div className="texts">
          <h2>
            Donâ€™t need it?
            <br />
            Giftify it!
          </h2>
          <p>
            Get rid of unused stuff clogging up your living space and give a
            hand to your neighbours by gifting them!
          </p>
        </div>
        <div className="heroImg">
          {/* an actual picture is needed */}
          <img alt="giftify" src="#" />
        </div>
      </div>
      {/*try the button target*/}
      <div className="giftifyButton">
        <Router>
          <Link to="/donation/add_product_card">
            <Button variant="primary">Giftify Now!</Button>
          </Link>
        </Router>
      </div>
    </div>
  );
}
export default Hero;
