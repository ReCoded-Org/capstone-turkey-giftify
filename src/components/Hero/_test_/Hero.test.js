import React from "react";
import Hero from "./../Hero";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("hero section renders corrrectly", () => {
  const tree = renderer.create(<Hero />).toJSON();
  expect(tree).toMatchSnapshot();
});
