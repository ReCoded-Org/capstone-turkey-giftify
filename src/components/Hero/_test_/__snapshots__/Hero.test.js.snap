// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="hero"
>
  <div
    className="upperPart"
  >
    <div
      className="texts"
    >
      <h1>
        Donâ€™t need it?
        <br />
        Giftify it!
      </h1>
      <p>
        Get rid of unused stuff clogging up your living space and give a hand to your neighbours by gifting them!
      </p>
    </div>
    <div
      className="heroImg"
    >
      <img
        alt="giftify"
        src="#"
      />
    </div>
  </div>
  <div
    className="giftifyButton"
  >
    <a
      href="/donation/add_product_card"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="button"
      >
        Giftify Now!
      </button>
    </a>
  </div>
</div>
`;
